hpx/traits/is_callable_with.hpp

```diff
@@ -0,0 +1,38 @@
+//  Copyright (c) 2016 Lukas Troska
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_TRAITS_IS_CALLABLE_WITH_HPP
+#define HPX_TRAITS_IS_CALLABLE_WITH_HPP
+
+#include <boost/type_traits/integral_constant.hpp>
+
+#include <type_traits>
+
+namespace hpx { namespace traits
+{
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        struct is_callable_with_impl
```

Comment:
Yes I have seen it. It just tests if the object f is callable at all. I need a trait that tells me if f is callable with f(std::size_t, future_type) or just f(future_type), since clearly it is callable in the first place.